src/arraymancer/tensor/aggregate.nim

```diff
@@ -245,3 +245,33 @@ proc iqr*[T](t: Tensor[T]): float =
   let tS = t.sorted
   result = percentile(tS, 75, isSorted = true) -
            percentile(tS, 25, isSorted = true)
+
+proc cumsum*[T](t: Tensor[T], axis:int): Tensor[T] = # from hugogranstrom
+  ## Calculates the cumulative sum of a rank-n Tensor.
+  ## Inputs:
+  ##  - t: a rank-n tensor to cumulatively sum
+  ##  - axis: int
+  ## Returns:
+  ##  - A tensor cumulatively summed at axis, that is, add each value to
+  result = zeros_like(t)
+  for i, tAxis in enumerateAxis(t, axis):
+    var temp = result.atAxisIndex(axis, i)
+    if i == 0:
+      temp[_] = tAxis
+    else:
+      temp[_] = result.atAxisIndex(axis, i-1) + tAxis
+
+proc cumprod*[T](t: Tensor[T], axis:int): Tensor[T] = # from hugogranstrom
+  ## Calculates the cumulative sum of a rank-n Tensor.
+  ## Inputs:
+  ##  - t: a rank-n tensor to cumulatively sum
+  ##  - axis: int
+  ## Returns:
+  ##  - A tensor cumulatively summed at axis, that is, add each value to
```

Comment:
and similar here can you add `mixin _` here, it seems like the `_` symbol is not visible in some cases and breaks CI

Reply:
added